src/pages/Swap/index.tsx

```diff
@@ -410,53 +410,59 @@ export function Swap({
     allowance.state === AllowanceState.ALLOWED ? allowance.permitSignature : undefined
   )
 
-  const handleSwap = useCallback(() => {
+  const handleSwap = useMemo(() => {
```

Comment:
This is changed to `useMemo` so that we don't have a nop callback when it's not ready to render, but we just have `null`, and can disable the button based on that value.